### PR TITLE
 Submitting CS-Script version 4.8.9

### PR DESCRIPTION
**Pull request by oleg-shilo**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/126892)

### CODE REVIEW

**Comment by stephengillie** [comment]
Manual Validation ended with:
> PS C:\Users\User\Desktop> C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\oleg-shilo.cs-script__DefaultSource\cscs.exe
> You must install .NET to run this application.
> 
> App: C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\oleg-shilo.cs-script__DefaultSource\cscs.exe
Architecture: x64
App host version: 8.0.0
.NET location: Not found
> 
> Learn more:
> https://aka.ms/dotnet/app-launch-failed
> 
> Download the .NET runtime:
https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=win-x64&os=win10&apphost_version=8.0.0

**Comment by stephengillie** [comment]
Hi @oleg-shilo,

This package appears to have a dependency. I've tried both `Microsoft.DotNet.AspNetCore.8` and `Microsoft.DotNet.Runtime.8`, but the application still gives this error when launched. Do you happen to know more about the dependency?

**Comment by oleg-shilo** [comment]
Hi Stephen, indeed the package has a dependency on .NET 8 
```
winget install Microsoft.DotNet.SDK.8
```
Though it `Microsoft.DotNet.DesktopRuntime.8` is present the the product can also function but only partially. 

The SDK presence is the dependency nominated in the product [description](https://github.com/oleg-shilo/cs-script).

I am not sure how we are handling the dependencies in the winget manifests. On Chocolatey I did not have to do that so I assumed similar expectations for winget. Should it be done differently?

**Comment by stephengillie** [comment]
Manual Validation ended with:
```PowerShell
PS C:\Users\User\Desktop> C:\Users\User\AppData\Local\Microsoft\WinGet\Links\cscs.exe
The application to execute does not exist: 'C:\Users\User\AppData\Local\Microsoft\WinGet\Links\cscs.dll'.
PS C:\Users\User\Desktop> C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\oleg-shilo.cs-script__DefaultSource\cscs.exe
C# Script execution engine (.NET Core). Version 4.8.9.0.
Copyright (C) 2004-
> 2023 Oleg Shilo. www.csscript.net (github.com/oleg-shilo/cs-script)

   CLR:             8.0.0
   System:          Microsoft Windows NT 10.0.19044.0
   Architecture:    x64
   Install dir:     <not integrated>
   Script engine:   C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\oleg-shilo.cs-script__DefaultSource\cscs.dll
   Compiler engine: dotnet (C:\Program Files\dotnet\dotnet.exe)
   NuGet manager:   dotnet
   NuGet cache:     <not found>
   Script cache:    C:\Users\User\AppData\Local\Temp\csscript.core\cache
   Custom commands: C:\ProgramData\cs-script\commands
   Global includes: C:\ProgramData\cs-script\inc

PS C:\Users\User\Desktop>
```

File: manifests/o/oleg-shilo/cs-script/4.8.9.0/oleg-shilo.cs-script.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Created using wingetcreate 1.5.7.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: oleg-shilo.cs-script
+PackageVersion: 4.8.9.0
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: cscs.exe
+  PortableCommandAlias: cscs
```
Comment (stephengillie) [inline manifests/o/oleg-shilo/cs-script/4.8.9.0/oleg-shilo.cs-script.installer.yaml:10]:
```suggestion
  PortableCommandAlias: cscs
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.DotNet.SDK.8
```
Apologies, I'm getting ahead of myself.

**Comment by stephengillie** [comment]
Hi @oleg-shilo,

After adding the dependency [here](https://github.com/microsoft/winget-pkgs/pull/126892#discussion_r1397847617), I received the output [here](https://github.com/microsoft/winget-pkgs/pull/126892#issuecomment-1817051250). This is a common issue we're working to resolve, where the application works normally, but the symlink is missing DLL files.

**Comment by oleg-shilo** [comment]
> ...where the application works normally, but the symlink is missing DLL files... 

Yeah I also found that. So I gave up on the idea of a link in the links folder and let user to work with the executable directly. It's possible because the location of the cscs.exe is in the PATH. Meaning that I do not really need the symlink.

image

I wouldn't mind having a link for an alias like `css` but that does not work because of the dependencies (as you also discovered). So I actually solved this by giving the user the option to create not a symbolic but a hard link in the location of the exe:

```txt
cscs -self-alias
```

And this executes in the background:
```
c:
cd C:\Users\olegs\AppData\Local\Microsoft\WinGet\Packages\oleg-shilo.cs-script__DefaultSource
mklink /H css.exe cscs.exe
```

And it produces a nice css.exe (alias===hard-link)
image

And now one can execute the script engine with the `css` alias:
image

> This is a common issue we're working to resolve,...
You can probably approve the manifest for merging (my app is not really impacted) and work on the issue without the pressure.

I also hope that that hard-link instead of a soft-link trick can potentially help you with your challenge. Good luck.

Thank you for your support.

**Comment by oleg-shilo** [comment]
> This is a common issue we're working to resolve

While WinGet team is working on it, would you mind to approve merging the PR?
It just happens that my product is no impacted by this winget issue so we do not need to wait till the solution is provisioned. 

Thank you.

**Comment by oleg-shilo** [comment]
It seems like this PR is not going anywhere. So I am closing it as the next version of CS-Script manifest has been successfully merged.

Guys, if you are reading it please consider using shims instead of symbolic links for executables from the portable packages. Chocolatey has gone through the same problem-resolution process and only solved it by moving to shims. You can also do it. You cannot use Chocolatey ShimGen as it is not licenced for that.

BTW you can even use MIT-licenced shim generator that I recently [published with you](https://github.com/microsoft/winget-pkgs/pull/127501). It solves this very problem you are facing.

**Pull request closed without merging**